### PR TITLE
Added the initial version of global scheduler startup script

### PR DESCRIPTION
**Pull request by jshaofuturewei**
Removed kube-controller-manager
TODO: Remove kube-proxy, kuberlet, and kube-scheduler

### CODE REVIEW

File: hack/lib/common.sh [hunk 1]
```diff
@@ -645,4 +645,13 @@ function kube::common::generate_kubeproxy_certs {
     fi
 }
 
-
+function kube::common::start_global_resource_scheduler {
+    CONTROLPLANE_SUDO=$(test -w "${CERT_DIR}" || echo "sudo -E")
+    kubeconfigfilepaths="${CERT_DIR}/controller.kubeconfig"
+    
+    GSR_LOG=${LOG_DIR}/global-resource-scheduler.log
+    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" kube-controller-manager \
```
Comment (CoderKevinZhang) [inline hack/lib/common.sh:653]: Is it temporary that we use kube-controller-manager for now?

Comment (jshaofuturewei) [reply hack/lib/common.sh:653]: Yes, it is used for cluster controllers now.

**Review by CoderKevinZhang — APPROVED** [review]
LGTM

File: hack/lib/common.sh (hunk 1 above)
Comment (jshaofuturewei) [reply hack/lib/common.sh:653]: The hq scheduler uses kube-controller-manager now. We might remove it in the future.

File: hack/lib/common.sh [hunk 2]
```diff
@@ -645,4 +645,13 @@ function kube::common::generate_kubeproxy_certs {
     fi
 }
 
-
+function kube::common::start_global_resource_scheduler {
```
[the remark this answers is not in the document]
Comment (jshaofuturewei) [reply hack/lib/common.sh:648]: This is what you sent me via email

function kube::common::start_global_resource_scheduler {
    CONTROLPLANE_SUDO=$(test -w "${CERT_DIR}" || echo "sudo -E")
    kubeconfigfilepaths="${CERT_DIR}/admin.kubeconfig"
    
    GSR_LOG=${LOG_DIR}/global-resource-scheduler.log
    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" globalscheduler \
      --v="${LOG_LEVEL}" \
      --kubeconfig "${kubeconfigfilepaths}"  >"${GSR_LOG}" 2>&1 &
    GSR_PID=$!
}

I updated it to the function kube::common::start_global_resource_scheduler
function kube::common::start_global_resource_scheduler {
    CONTROLPLANE_SUDO=$(test -w "${CERT_DIR}" || echo "sudo -E")
    kubeconfigfilepaths="${CERT_DIR}/admin.kubeconfig"
    
    GSR_LOG=${LOG_DIR}/global-resource-scheduler.log
    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" globalscheduler \
      --v="${LOG_LEVEL}" \
      --kubeconfig "${kubeconfigfilepaths}"  >"${GSR_LOG}" 2>&1 &
    GSR_PID=$!
}

[the remark this answers is not in the document]
Comment (kimeunju108) [reply hack/lib/common.sh:648]: According to the meeting with Jun & Eunju at 9:30 am 12-1-2020, the following code is for running cluster controller.
function kube::common::start_global_resource_scheduler {
    CONTROLPLANE_SUDO=$(test -w "${CERT_DIR}" || echo "sudo -E")
    kubeconfigfilepaths="${CERT_DIR}/admin.kubeconfig"
    
    GSR_LOG=${LOG_DIR}/global-resource-scheduler.log
    ${CONTROLPLANE_SUDO} "${GO_OUT}/hyperkube" globalscheduler \
      --v="${LOG_LEVEL}" \
      --kubeconfig "${kubeconfigfilepaths}"  >"${GSR_LOG}" 2>&1 &
    GSR_PID=$!
}

**Review by kimeunju108 — APPROVED** [review]
Jun and Eunju had a short meeting to resolve a review question.

**Pull request merged**